applications/KZero.md

```diff
@@ -0,0 +1,300 @@
+| Ethereum | ❓                | https://ethresear.ch/t/zklogin-framework-based-on-4337-contract-wallet/20855 | [Privy](https://x.com/0xJamesXXX/status/1831663721024012621)[Dynamic](https://app.dynamic.xyz/) |
+| Sui      | ✅                | https://sui.io/zklogin                                       | [Sui Wallet](https://suiwallet.com/)                         |
+| Starknet | ✅                | https://mirror.xyz/xarlabs.eth/kfUf0XyK-sxig3DYBKvVcr836HEyvkqqxU0uN-tjAHo | [zKey](https://www.zkey.org/)                                |
+| Aptos    | ✅                | https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-61.md | [Keyless Wallet](https://aptos.dev/en/build/guides/aptos-keyless) |
+
+Some of the products attract millions of people, [like Privy](https://bingx.com/zh-tc/support/articles/36563397225497/), is important to the Polkadot ecosystem as well.
+
+
+
+## Highlights of KZero
+
+- **Keyless:** Say goodbye to managing private keys. One-click access to your DAppswithout keys or device dependency. Simple, fast, and frictionless.
+- **100% Self-Custodial:** Users maintain complete ownership of their assets — your data stays with you. Maximum security for your assets and identity, always under your control. 
+- **Maximum Security Assurance:** Kzero introduces a revolutionary approach to key management, empowering users to seamlessly change their local temporary private keys at their convenience while providing the option to set expiration times for each key. 
+- **Simplified Onboarding:** One-click seamless experience. Users can initiate the onboarding process with a single click by choosing to "Login with Google," after that, users can easily start to submit transactions.
+- **ZKP-based Social Login.** **Privacy Protected:** Zero-knowledge proof (ZKP) lets users interact with Web3 while keeping their data private. No compromises—access DApps securely and anonymously.
+- **Multiple** **JWT** **providers:** Users can login with Google, Twitch, Kakao, Facebook, Apple, Slack and other OpenID Providers.
+
+
+
+## Tech Details
+
+### How Kzero works
+
+![zklogin-mechanism](https://p.ipic.vip/uicx29.png)
+
+(Step 0) We plan to use Groth16 for our protocol’s zkSNARK instantiation, requiring a singular generation of a structured common reference string (CRS) linked to the circuit. A ceremony is conducted to generate the CRS, which is used to produce the proving key in the ZK Proving Service, the verifying key in Project Authority.
+
+(Step 1-3) The user begins by logging into an OpenID Provider (OP) to obtain a JWT token containing a defined nonce. In particular, the user generates an ephemeral KeyPair (eph_sk, eph_pk), along with expiry times (max_epoch) and randomness (jwt_randomness). Through these three parameters, the Wallet Extension can thus compute the nonce. After the user completes the OAuth login flow with the nonce, an JWT token can be retrieved.
```

Comment:
Could you elaborate on how the randomness is generated?

Reply:
Add this content in a new section titled "Security and Privacy"

Reply:
Add this content in a new section titled "Security and Privacy"